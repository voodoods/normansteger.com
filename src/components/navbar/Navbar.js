import React, { Component } from 'react'
import PropTypes from 'prop-types'
import StyledNavbar from './Navbar.styled'
import items from './Navbar.items.json'

class Navbar extends Component {
  render() {
    return (
      <StyledNavbar className="Navbar pure-u-lg-1 pure-u-md-1 pure-u-sm-1">
        <ul className="pure-u-lg-1 pure-u-md-1 pure-u-sm-1">
          {items.map((item, index) => (
            <li key={index} className="pure-u-lg-1 pure-u-md-1 pure-u-sm-1">
              <a href={item.route} className={this.props.activePath === item.route ? 'active' : 'inactive'}>
                {item.title}
              </a>
            </li>
          ))}
        </ul>
      </StyledNavbar>
    )
  }
}

Navbar.propTypes = {
  items: PropTypes.array,
  activePath: PropTypes.string,
}

export default Navbar
