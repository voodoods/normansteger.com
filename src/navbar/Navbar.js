import React, { Component } from 'react'
import StyledNavbar from './Navbar.styled'

class Navbar extends Component {
  render() {
    return (
      <StyledNavbar className="Navbar">
        <ul>
          <li>
            <a href="/skills">My Professions</a>
          </li>
          <li>
            <a href="/about-me">About me</a>
          </li>
          <li>
            <a href="/hire-me">Hire me</a>
          </li>
        </ul>
      </StyledNavbar>
    )
  }
}

export default Navbar
