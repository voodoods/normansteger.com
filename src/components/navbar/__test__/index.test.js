import React from 'react'
import { shallow } from 'enzyme'
import Navbar from '../'

describe('<Navbar /> component index', () => {
  it('expects component to match snapshot', () => {
    const wrapper = shallow(<Navbar />)
    expect(wrapper).toMatchSnapshot()
  })
})
