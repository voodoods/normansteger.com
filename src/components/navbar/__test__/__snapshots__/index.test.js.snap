// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> component index expects component to match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navbar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <ul
        className="pure-u-lg-1 pure-u-md-1 pure-u-sm-1"
>
        <li
                className="pure-u-lg-1 pure-u-md-1 pure-u-sm-1"
        >
                <a
                        className="inactive"
                        href="/"
                >
                        Home
                </a>
        </li>
        <li
                className="pure-u-lg-1 pure-u-md-1 pure-u-sm-1"
        >
                <a
                        className="inactive"
                        href="/skills"
                >
                        Skills
                </a>
        </li>
        <li
                className="pure-u-lg-1 pure-u-md-1 pure-u-sm-1"
        >
                <a
                        className="inactive"
                        href="/cv"
                >
                        Curriculum Vitae
                </a>
        </li>
</ul>,
      "className": "Navbar pure-u-lg-1 pure-u-md-1 pure-u-sm-1",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="pure-u-lg-1 pure-u-md-1 pure-u-sm-1"
>
            <a
                        className="inactive"
                        href="/"
            >
                        Home
            </a>
</li>,
          <li
            className="pure-u-lg-1 pure-u-md-1 pure-u-sm-1"
>
            <a
                        className="inactive"
                        href="/skills"
            >
                        Skills
            </a>
</li>,
          <li
            className="pure-u-lg-1 pure-u-md-1 pure-u-sm-1"
>
            <a
                        className="inactive"
                        href="/cv"
            >
                        Curriculum Vitae
            </a>
</li>,
        ],
        "className": "pure-u-lg-1 pure-u-md-1 pure-u-sm-1",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="inactive"
              href="/"
>
              Home
</a>,
            "className": "pure-u-lg-1 pure-u-md-1 pure-u-sm-1",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Home",
              "className": "inactive",
              "href": "/",
            },
            "ref": null,
            "rendered": "Home",
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="inactive"
              href="/skills"
>
              Skills
</a>,
            "className": "pure-u-lg-1 pure-u-md-1 pure-u-sm-1",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Skills",
              "className": "inactive",
              "href": "/skills",
            },
            "ref": null,
            "rendered": "Skills",
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="inactive"
              href="/cv"
>
              Curriculum Vitae
</a>,
            "className": "pure-u-lg-1 pure-u-md-1 pure-u-sm-1",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Curriculum Vitae",
              "className": "inactive",
              "href": "/cv",
            },
            "ref": null,
            "rendered": "Curriculum Vitae",
            "type": "a",
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <ul
          className="pure-u-lg-1 pure-u-md-1 pure-u-sm-1"
>
          <li
                    className="pure-u-lg-1 pure-u-md-1 pure-u-sm-1"
          >
                    <a
                              className="inactive"
                              href="/"
                    >
                              Home
                    </a>
          </li>
          <li
                    className="pure-u-lg-1 pure-u-md-1 pure-u-sm-1"
          >
                    <a
                              className="inactive"
                              href="/skills"
                    >
                              Skills
                    </a>
          </li>
          <li
                    className="pure-u-lg-1 pure-u-md-1 pure-u-sm-1"
          >
                    <a
                              className="inactive"
                              href="/cv"
                    >
                              Curriculum Vitae
                    </a>
          </li>
</ul>,
        "className": "Navbar pure-u-lg-1 pure-u-md-1 pure-u-sm-1",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="pure-u-lg-1 pure-u-md-1 pure-u-sm-1"
>
              <a
                            className="inactive"
                            href="/"
              >
                            Home
              </a>
</li>,
            <li
              className="pure-u-lg-1 pure-u-md-1 pure-u-sm-1"
>
              <a
                            className="inactive"
                            href="/skills"
              >
                            Skills
              </a>
</li>,
            <li
              className="pure-u-lg-1 pure-u-md-1 pure-u-sm-1"
>
              <a
                            className="inactive"
                            href="/cv"
              >
                            Curriculum Vitae
              </a>
</li>,
          ],
          "className": "pure-u-lg-1 pure-u-md-1 pure-u-sm-1",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": <a
                className="inactive"
                href="/"
>
                Home
</a>,
              "className": "pure-u-lg-1 pure-u-md-1 pure-u-sm-1",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Home",
                "className": "inactive",
                "href": "/",
              },
              "ref": null,
              "rendered": "Home",
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": <a
                className="inactive"
                href="/skills"
>
                Skills
</a>,
              "className": "pure-u-lg-1 pure-u-md-1 pure-u-sm-1",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Skills",
                "className": "inactive",
                "href": "/skills",
              },
              "ref": null,
              "rendered": "Skills",
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": <a
                className="inactive"
                href="/cv"
>
                Curriculum Vitae
</a>,
              "className": "pure-u-lg-1 pure-u-md-1 pure-u-sm-1",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Curriculum Vitae",
                "className": "inactive",
                "href": "/cv",
              },
              "ref": null,
              "rendered": "Curriculum Vitae",
              "type": "a",
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
